server_test.go

```diff
@@ -1,6 +1,112 @@
 package main
 
-import "testing"
+import (
+	"testing"
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"net"
+	"time"
+	"github.com/stretchr/testify/require"
+	"encoding/hex"
+	"github.com/lightningnetwork/lnd/brontide"
+	"github.com/roasbeef/btcd/connmgr"
+	"fmt"
+	"strings"
+	"io"
+)
+
```

Comment:
👍 for the start of unit tests for the sever! Sorely needed. Eventually we also need to extract the `peer` struct to its own package to allow us to better test it in isolation, also provide a library that people can use to connect to nodes in isolation.

Reply:
I may be interested in taking up those tasks in the near future if you can provide some guidance on requirements for the library.